fvm/evm/stdlib/contract.go

```diff
@@ -97,6 +99,48 @@ func (e abiDecodingError) Error() string {
 	return b.String()
 }
 
+func calculateComputation(
+	inter *interpreter.Interpreter,
+	values *interpreter.ArrayValue,
+	evmAddressTypeID common.TypeID,
+) int {
+	computation := 0
+
+	values.Iterate(inter, func(element interpreter.Value) (resume bool) {
+		switch value := element.(type) {
+		case *interpreter.StringValue:
+			// Dynamic variables, such as strings, are encoded
+			// in 3 chunks of 32 bytes. The first chunk contains
+			// the index where information for the string begin,
+			// the second chunk contains the number of bytes the
+			// string occupies, and the third chunk contains the
+			// value of the string itself.
+			computation += 3 * abiEncodingByteSize
```

Comment:
it seems then like the last chunk can be bigger than the byte size defined here?

Reply:
Ah, great catch :100: . 3 is the minimum number of chunks for a `String`. A `String` value with 50 characters, will need 2 extra chunks, because it always has to be in multiples of 32.

Reply:
Fixed in https://github.com/onflow/flow-go/pull/5146/commits/5a6dfa5545d0dfebd4258dbebe9319c930c4d5bf.
I will add some tests for strings & arrays that exceed the chunk size of 32 bytes.

Reply:
Tests added in https://github.com/onflow/flow-go/pull/5146/commits/32e6b6f0b533f8de085671e5fff80c4e4587118f